docs/build/getting-started/quickstarts/testing.md

```diff
@@ -0,0 +1,102 @@
+Testing is an essential part of smart contract development to ensure the correctness and reliability of your code. In Cadence, the testing framework provides a convenient way to write tests for Cadence programs, allowing you to verify the functionality of your smart contracts.
+
+## Install Flow CLI
+
+The [Flow CLI](../../../tools/flow-cli/index.md) is the primary tool for developing, testing, and deploying smart contracts written in Cadence to the Flow network. You can install the Flow CLI by following the [installation instructions](../../../tools/flow-cli/install.md) provided by the Flow documentation.
+
+## Create a new project
+
+In your preferred code editor, create a new directory for your project and navigate to it in the terminal. Then 
+initialize a new Flow project by running the command `flow init`. This will create a `flow.json` file that contains the project configuration.
+```bash
+mkdir test-cadence
+cd test-cadence
+flow init
+```
+
+## Write a simple smart contract
+
+In your code editor, create a new file called `calculator.cdc` and add the following code:
+
+```cadence calculator.cdc
+access(all) contract Calculator {
+    access(all) fun add(a: Int, b: Int): Int {
+        return a + b;
+    }
+
+    access(all) fun subtract(a: Int, b: Int): Int {
+        return a - b;
+    }
+}
+```
+
+## Add the smart contract to the config
+
+Run `flow config add contract` and add the created `calculator.cdc` contract to the config with the name as `Calculator`:
+
+```
+Enter name: Calculate
+Enter contract file location: ./calculator.cdc
+```
+
+## Write test cases
+
+In the same directory, create a new file called `calculator_test.cdc` and add the following code:
+
+```cadence calculator_test.cdc
+import Test
+import "Calculator" // contract name from the previous step
+
+access(all) let calculator = Calculator()
+
+access(all) fun testAdd() {
+  Test.assertEqual(calculator.add(a: 2, b: 3), 5)
+}
+
+access(all) fun testSubtract() {
+  Test.assertEqual(calculator.subtract(a: 5, b: 3), 2)
+}
+
+```
+
+This code
+- imports the `Calculator` contract from the `calculator.cdc` file (according to `flow.json`); 
+- creates an `calculator` instance of the smart-contract;
+- defines two test functions: `testAdd()` and `testSubtract()`;
+- calls `add()` and `subtract()` methods with different input values  respectively.
+
+## Running the test cases
+
+To run the test cases, use the following command in the terminal:
+
+```bash
+flow test ./calculator_test.cdc
+```
+
+This command uses the Flow CLI to run the test cases and display the output. You should see the following output:
+
+```
+Test results: "./calculator_test.cdc"
+- PASS: testAdd
+- PASS: testSubtract
+```
+
+This output indicates that both test cases ran successfully, and the smart contract is functioning as expected.
+
+## Advanced Testing Techniques
+
+The Cadence testing framework provides various features and techniques for writing comprehensive test cases. Some of these include:
```

Comment:
Might be some updates worth mentioning here
https://forum.onflow.org/t/major-uplift-for-cadence-testing-framework/5232

Reply:
This is very close to coming to life, https://github.com/onflow/flow-cli/pull/1227 just waiting for this :pray: